src/mlpack/tests/ann/layer/grouped_convolution.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * @file tests/ann/layer/grouped_convolution.cpp
+ * @author Shubham Agrawal
+ *
+ * Tests the ann layer modules.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+
+#include "../../test_catch_tools.hpp"
+#include "../../catch.hpp"
+#include "../../serialization.hpp"
+#include "../ann_test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::ann;
+
+/**
+ * Simple test for the Grouped Convolution layer (With groups = 2).
```

Comment:
Thanks for adding the clarifying comment here.  I still think it would be useful to add a test ensuring that `GroupedConvolution` with `groups = 1` is equivalent to `Convolution`, and perhaps another test for a larger value of `groups` (perhaps a handful of values for `groups`, up to you).